### ISSUE TITLE
FFI defined `properties` show incorrect information in the docs.

### ISSUE THREAD

**Issue by vicky5124** [mitmproxy/pdoc#300]
#### Problem Description
When a property is defined over FFI, and pdoc is used to build documentation out of the resulting dynamic library, the property is shown with incorrect information.

![11 24 03_18-09-2021](https://user-images.githubusercontent.com/8344197/133884032-a92afe04-6461-437c-9f55-59f2d786e436.png)

- It shows as an `attribute` instead of a `property` or `member`, even though attributes are different* in FFI.
- It shows that the module it's from comes from `builtins` instead of the module name.

*See questions 2

#### Questions:
1. To temporarily fix this: Is the value displayed on the docs customizable with some kind of `__dunder__` method or flag?
2. Are attributes actually different from properties on the resulting binary? Because in rust, they are defined differently, attributes are functions without any arguments (not even self), but properties are getters and setters that can take arguments and even raise exceptions.

#### Steps to reproduce the behaviour:
`Cargo.toml`
```toml
[package]
name = "pyo3_pdoc"
version = "0.1.0"
edition = "2018"

[lib]
name = "pyo3_pdoc"
crate-type = ["cdylib"]

[dependencies]
pyo3 = "0.14"
```
`src/lib.rs`
```rust
use pyo3::prelude::*;

#[pyclass]
struct Test {
    num: i32
}

#[pymethods]
impl Test {
    #[new]
    fn new(num: i32) -> Self {
        Self { num }
    }

    #[getter]
    fn num(&self) -> i32 {
        self.num
    }
}

#[pymodule]
fn pyo3_pdoc(_py: Python, m: &PyModule) -> PyResult<()> {
    m.add_class::<Test>()?;

    Ok(())
}
```

- Install rust: https://www.rust-lang.org/learn/get-started
- Start a venv instance `python -m venv .env`
- Source env vars `source .env/bin/activate`
- Install maturing `maturin develop`
- Build with `maturin develop`
- Generate docs: `pdoc -o docs .env/lib/python3.9/site-packages/pyo3_pdoc`

#### System Information
pdoc: 7.4.0
Python: 3.9.7
Platform: Linux-5.14.3-zen1-1-zen-x86_64-with-glibc2.33

**Comment by mhils** [comment]
Thanks for the detailed issue description! :cake: You'll get a longer reply as I first needed to figure things out as well. 😅 

> Are attributes actually different from properties on the resulting binary? Because in rust, they are defined differently, attributes are functions without any arguments (not even self), but properties are getters and setters that can take arguments and even raise exceptions.

For pdoc's purposes, we only distinguish between *modules, classes, functions,* and *variables*. That's a great simplification over what Python has internally, the intention here is to keep things relatively straightforward for the purpose of documentation. For practical purposes, we treat `@property` members as variables.
You can find similar behavior in Python's own documentation, where you can only see that it's a property when it's mentioned in the docstring (see https://docs.python.org/3/library/urllib.request.html#urllib.request.Request.full_url for example).


> Are attributes actually different from properties on the resulting binary?

Consider the following Python file:
```python
class TestPy:
    @property
    def num(self):
        return 42
```
This creates a TestPy class with a `num` attribute that is a property object:
```python
>>> TestPy.num
<property object at 0x000002853EC21B30>
>>> type(TestPy.num)
<class 'property'>
```

pdoc detects that it's a property and has a [special code path](https://github.com/mitmproxy/pdoc/blob/d60967e6b91cc1ef1dd057382beae9162c8c264d/pdoc/doc.py#L239-L264) that handles everything accordingly. Unfortunately your PyO3 module above generates a slightly different object:
```python
>>> from pyo3_pdoc import Test
>>> Test.num
<attribute 'num' of 'builtins.Test' objects>
>>> type(Test.num)
<class 'getset_descriptor'>
```

The `Test` class here has a `num` attribute that is a GetSet descriptor, which is basically a "`@property` for extension modules" (https://docs.python.org/3/howto/descriptor.html). Technically speaking that's (like `@property`) just an attribute, which we treat like a variable. We do [detect that it's a data descriptor](https://github.com/mitmproxy/pdoc/blob/d60967e6b91cc1ef1dd057382beae9162c8c264d/pdoc/doc.py#L275-L276) (which is why we are able to get the docstring), but other than that we still treat it like a variable.

Long story short, I think the correct fix here is to not display a default value for data descriptors. I'll send a PR in a minute. 😄 


> To temporarily fix this: Is the value displayed on the docs customizable with some kind of __dunder__ method or flag?

You can [edit pdoc's HTML template](https://pdoc.dev/docs/pdoc.html#editing-pdocs-html-template), which provides some very flexible means to customize the output. For example, you could override the [`default_value`](https://github.com/mitmproxy/pdoc/blob/d60967e6b91cc1ef1dd057382beae9162c8c264d/pdoc/templates/default/module.html.jinja2#L543-L547) macro here to hide it. But I'll ship a release in a bit anyhow. :)

**Comment by mhils** [comment]
pdoc 8 is out in a few minutes and has the fix for this. 😄
